### PR TITLE
Support dot-like method calls; alternative to #19384 #19919; fixes #19392

### LINKED ISSUE #19392: A macro with `-d:nimPreviewDotLikeOps` enabled gets different arguments
When including `-d:nimPreviewDotLikeOps` a dot like operator macro gets different arguments.

### Example
```nim
import std/macros
macro `?`(a: varargs[untyped]): untyped =
  echo a.treerepr

foo?bar(1)

macro `.?`(a: varargs[untyped]): untyped =
  echo a.treerepr

foo.?bar(1)
```

### Current Output

```
ArgList
  Ident "foo"
  Call
    Ident "bar"
    IntLit 1
ArgList
  Ident "foo"
  Ident "bar"
```

### Expected Output
```
ArgList
  Ident "foo"
  Call
    Ident "bar"
    IntLit 1
ArgList
  Ident "foo"
  Call
    Ident "bar"
    IntLit 1
```

### Additional Information
Tested in devel HEAD.

```
Nim Compiler Version 1.7.1 [Windows: amd64]
Compiled at 2022-01-15
Copyright (c) 2006-2022 by Andreas Rumpf

git hash: a95399143fdbd518f4d5fe33487b656c4cde7d6d
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by ringabout**
closes #19384 
closes #19919
fixes #19392

tests are coming ...

### CODE REVIEW

**Comment by ringabout** [comment]
I'm checking https://github.com/nim-lang/Nim/issues/19806

**Comment by ringabout** [comment]
Hi @markspanbroek. I tested the tests of question, all of them pass. I would like to add questionable to important packages in case of regression. However it is blocked by https://github.com/mratsim/weave/pull/182

**Comment by ringabout** [comment]
```nim
for i in x .++ (if true: 12):
  collector.add i
```
if expr gives the same error like before:

```
Error: expression '12' is of type 'int literal(12)' and has to be used (or discarded)
```

File: tests/misc/trfc_341_dotlike_call.nim [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  var res: T = a
+  while res <= b:
+    yield res
+    inc res
+    inc res
+
+block:
+  var collector: seq[int] = @[]
+  for i in 0 .++ 8:
+    collector.add i
```
Comment (ringabout) [inline tests/misc/trfc_341_dotlike_call.nim:40]: Edge case:
```nim
for i in x .++ y + 3:
  collector.add i
```
I'm thinking of it.

Comment (ringabout) [reply tests/misc/trfc_341_dotlike_call.nim:40]: It seems that first I need to parse expr, then check whether it is a symbol.

File: compiler/parser.nim [hunk 2]
```diff
@@ -479,7 +482,19 @@ proc dotLikeExpr(p: var Parser, a: PNode): PNode =
...
+    if p.tok.tokType == tkParLe and p.tok.strongSpaceA <= 0:
+      let callNode = newNodeI(nkCall, p.parLineInfo)
+      callNode.add symbol
+      exprColonEqExprListAux(p, tkParRi, callNode)
+      result.add callNode
+    else:
+      result.add symbol
+      result = parseGStrLit(p, result)
+  else:
+    result.add simpleExpr(p)
```
Comment (xflywind) [inline compiler/parser.nim:497]: Or parMessage? Disallow defining dot like operators without symbols.

```nim
proc `.++`[T](a: T, b: T): T
```

**Comment by Araq** [comment]
Please explain your solution here.

**Comment by ringabout** [comment]
Sorry for the delay. My PR has the same problem as https://github.com/nim-lang/Nim/issues/8759 ([parser] a => counter+=1 parsed as (a => counter) += 1, violating the spec, and causing issues with => )

**Comment by Menduist** [comment]
What's the status on this PR :) ? Looking forward for questionable to work again on devel

**Comment by ringabout** [comment]
@Menduist, Sorry for the inconveneience. I cannot think of a good solution.

https://github.com/nim-lang/Nim/pull/19919 can be merged now which reverts enabling the experiemental feature, we can embrace dot-like method calls in the future.

**Comment by Menduist** [comment]
Maybe it's the best solution indeed, we can keep it as an optional flag until we find a solution / give up & remove it

**Comment by ringabout** [comment]
I will revisit it later.

**Pull request closed without merging**